### PR TITLE
Stormpath authentication support

### PR DESCRIPTION
**Pull request by mmoayyed**
Closes https://github.com/pac4j/pac4j/issues/182

### CODE REVIEW

File: pac4j-stormpath/src/main/java/org/pac4j/stormpath/credentials/StormpathAuthenticator.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import org.pac4j.stormpath.profile.StormpathProfile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An authentication handler for <a href="http://www.stormpath.com">Stormpath</a>.
+ * <p/>
```
Comment (leleuj) [inline pac4j-stormpath/src/main/java/org/pac4j/stormpath/credentials/StormpathAuthenticator.java]: This makes the builds fail (when generating Javadoc)::

```
[ERROR] Exit code: 1 - /home/travis/build/pac4j/pac4j/pac4j-stormpath/src/main/java/org/pac4j/stormpath/credentials/StormpathAuthenticator.java:21: error: self-closing element not allowed
[ERROR] * <p/>
[ERROR] ^
```

File: pac4j-stormpath/src/main/java/org/pac4j/stormpath/profile/StormpathProfile.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.pac4j.stormpath.profile;
+
+import org.pac4j.core.profile.CommonProfile;
+
+/**
+ * <p>The user profile returned from a Stormpath Authentication event.</p>
+ * @author Misagh Moayyed
+ * @since 1.8
+ */
+public class StormpathProfile extends CommonProfile {
```
Comment (leleuj) [inline pac4j-stormpath/src/main/java/org/pac4j/stormpath/profile/StormpathProfile.java]: We could maybe go further here by overriding the appropriate methods of the `CommonProfile` and by getting the appropriate properties. Maybe the `getDisplayName` should return the "fullName" attribute, the `getUsername` method should return the "surName"...

**Comment by leleuj** [comment]
Excellent! +1 with two comments

**Comment by leleuj** [comment]
One more think: don't forget to update the README.

**Comment by mmoayyed** [comment]
Updated pull.

**Comment by leleuj** [comment]
Thanks

**Comment by mmoayyed** [comment]
Was about to update the readme, but looks like you beat me to it. Thanks!

**Pull request merged**